### DISCUSSION TITLE
Clustering algorithm for calculating cross-axis tilt zones

### DISCUSSION THREAD

**Discussion by kurt-rhee** [pvlib/pvlib-python#1533] [Ideas]
A common method for modeling projects on terrain which has multiple cross-axis tilts is to create zones where each zone has it's own mono-slope backtracking algorithm.  I think that an open question is how many zones might be needed to closely approximate a terrain adaptive backtracking strategy which takes into account all of the cross-axis tilts natively.  This number of zones will increase or decrease depending on the complexity of the terrain.

I am working on using clustering algorithms to define which trackers should be included in what zones.  If this is something that others might find useful I am happy to contribute the results to pvlib.

**Comment by cwhanse** [comment]
I've had a few conversations about interest in modeling PV systems on uneven terrain, and what that might take. The clustering algorithm sounds like one of several pieces that could make that possible.

**Comment by kandersolar** [comment]
Sounds interesting! Also sounds nontrivial enough that it would need a published reference for a pvlib implementation to point to.
